stellargraph/data/explorer.py

```diff
@@ -789,3 +793,136 @@ def _check_neighbourhood_sizes(self, in_size, out_size):
+                * "exponential" - heavily biased towards more recent edges.
+            walk_bias (str, optional): distribution to use when choosing a random
+                neighbour to walk through. Available options are:
+                * None (default) - neighbours are picked from a uniform distribution
+                * "exponential" - exponentially decaying probability, resulting in a bias
+                    towards shorter time gaps
+            seed (int, optional): Random number generator seed; default is None
+
+        Returns:
+            List of lists of nodes ids for each of the random walks
+
+        """
+        self._check_common_parameters(nodes, n, length, seed)
+        np_rs = self._np_random_state if seed is None else np.random.RandomState(seed)
+
+        walks = []
+        for node in nodes:  # iterate over root nodes
+            for walk_number in range(n):  # generate n walks per root node
+                walk = self._walk(
+                    node, length, bidirectional, initial_edge_bias, walk_bias, np_rs
+                )
+                walks.append(walk)
+        return walks
+
+    def _exp_biases(self, times, t_0, decay):
+        # t_0 assumed to be smaller than all time values
+        dist = np.exp(t_0 - np.array(times) if decay else np.array(times) - t_0)
+        sum_dist = np.sum(dist)
+        return dist / sum_dist
+
+    def _biases(self, times, t_0, bias_type, is_forward):
+        if bias_type == "exponential":
+            # exponential decay bias needs to be reversed if looking backwards in time
+            return self._exp_biases(times, t_0, decay=is_forward)
+        else:
+            raise ValueError("Unsupported bias type")
+
+    def _step(self, node, time, is_forward, bias_type, np_rs):
+        """
+        Perform 1 temporal step from a node. Returns None if a dead-end is reached.
+
+        """
+
+        def check_time(t):
+            # time is None indicates no temporal filtering (i.e. initial edge selection)
+            return (
+                (time is None)
+                or (is_forward and t > time)
+                or (not is_forward and t < time)
+            )
+
+        neighbours = [
+            (neighbour, t)
+            for neighbour, t in self.graph.neighbors(node, include_edge_weight=True)
+            if check_time(t)
+        ]
+
+        if len(neighbours) > 0:
+            if bias_type is not None:
+                times = [t for _, t in neighbours]
+
+                # time is None indicates we should obtain the minimum available time for t_0
+                t_0 = time if time is not None else min(times)
+                biases = self._biases(
+                    times, t_0, bias_type=bias_type, is_forward=is_forward
+                )
+            else:
+                # default to uniform random sampling
+                biases = None
+            chosen_index = np_rs.choice(len(neighbours), p=biases)
```

Comment:
I think this will be faster using `naive_weighted_choices` (in this file).

```suggestion
            chosen_index = naive_weighted_choices(np_rs, biases)
```

although that doesn't handle `biases == None` (at the moment).

Reply:
Hmm I guess I can do
```suggestion
            chosen_index = naive_weighted_choices(np_rs, biases) if biases is not None else np_rs.choice(len(neighbours))
```
I think it's a bit awkward to update `naive_weighted_choices` to handle `biases==None` since then you need another parameter to specify the length